lib/client/sso.go

```diff
@@ -66,6 +67,8 @@ func (tc *TeleportClient) ssoLoginInitFn(keyRing *KeyRing, connectorID, connecto
 			return "", trace.Wrap(err)
 		}
 
+		codeVerifier := oauth2.GenerateVerifier()
```

Comment:
As we discussed previously offline, this PKCE code verifier logic makes the existing ["secret_key" logic](https://github.com/gravitational/teleport/blob/c7d467afdaade8d24f658be105e77fa846768d40/lib/client/sso/redirector.go#L234) redundant. Can you add a TODO (with my username if you'd like) to remove this dead code in the future?

For context for others, the pre-existing "secret_key" creates trust between the Teleport client and auth/proxy server for CLI clients only. The new PKCE flow takes it a step further by creating trust between the Teleport client, auth/proxy server, _and_ the OIDC provider, while also extending support for Web sessions.